### PR TITLE
`Development`: Update client dependencies

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.

### Motivation and Context
Update to the latest dependencies

### Steps for Testing
Try out a quiz creation and make sure changing the duration works correctly with different values

#### Exam Mode Testing
Try out a quiz creation and make sure changing the duration works correctly with different values

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Test Coverage
no changes

### Screenshots
no changes

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/team/team-participate/team-students-online-list.component.ts [hunk 1]
```diff
@@ -134,9 +134,12 @@ export class TeamStudentsOnlineListComponent implements OnInit, OnDestroy {
         });
         if (this.typingTeamStudents.length > 0) {
             const lastTypingDates = this.typingTeamStudents.map((student: OnlineTeamStudent) => student.lastTypingDate).filter(Boolean);
-            const earliestExpiration = dayjs.min(lastTypingDates).add(this.showTypingDuration, 'ms');
-            const timeToExpirationInMilliseconds = earliestExpiration.diff(dayjs());
-            setTimeout(() => this.computeTypingTeamStudents(), timeToExpirationInMilliseconds);
+            const minTypingDate = dayjs.min(lastTypingDates);
+            if (minTypingDate != undefined) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/team/team-participate/team-students-online-list.component.ts:138]:
```suggestion
            if (minTypingDate) {
```

File: src/main/webapp/app/complaints/complaint.service.ts [hunk 2]
```diff
@@ -260,7 +260,12 @@ export class ComplaintService implements IComplaintService {
         }
         const complaintStartDate = dayjs.max(possibleComplaintStartDates);
 
-        return dayjs().isBefore(complaintStartDate) ? undefined : complaintStartDate.add(complaintTimeFrame, 'days');
+        if (dayjs().isBefore(complaintStartDate)) {
+            return undefined;
+        }
+        if (complaintStartDate != undefined) {
```
Comment (MaximilianAnzinger) [inline src/main/webapp/app/complaints/complaint.service.ts:266]:
```suggestion
        if (complaintStartDate) {
```

Comment (Strohgelaender) [reply src/main/webapp/app/complaints/complaint.service.ts:266]: I'm a bit confused about this place in general. If `complaintStartDate` actually is undefined, the call above would not make much sense (`dayjs().isBefore(undefined)` always evaluates to false). So we don't return `undefined` above, but also don't return the value inside this case. In the end we implicitly return undefined, but without calling any return statement.

Comment (krusche) [reply src/main/webapp/app/complaints/complaint.service.ts:266]: feel free to propose an alternative. The method can return undefined and I would not worry too much about 

`if (complaintStartDate != undefined)` vs `if (complaintStartDate)`

the dayjs API changed and max can now also return undefined so this change was necessary

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**